app/uk/gov/hmrc/calculatenifrontend/controllers/TablesController.scala

```diff
@@ -55,12 +63,102 @@ class TablesController @Inject()(
 
     val selectedCategory = category.filter(categories.contains).getOrElse(categories.head)
 
-    val filteredBand: List[(String, RateDefinition)] =
-      band.filter{
-        case (_, v) => (v.employee.keySet ++ v.employer.keySet).contains(selectedCategory)
-      }.toList.sortBy(_._2.year.lowerValue.get)
-
+    val filteredBand: List[(String, RateDefinition)] = band.filter{
+      case (_, v) => (v.employee.keySet ++ v.employer.keySet).contains(selectedCategory)
+    }.toList.sortBy(_._2.year.lowerValue.get)
 
     Ok(classOnePage(selectedInterval, intervals, selectedCategory, categories, filteredBand))
   }
+
+  def classTwo(
+    date: Option[LocalDate]
+  ): Action[AnyContent] = Action { implicit request =>
+    val intervals: List[(LocalDate, String)] = intervalDropdown(ni.classTwo)
+    val dateP = date.getOrElse(intervals.head._1)
+
+    ni.classTwo.at(dateP) match {
+      case Some(data) =>
+        val selectedInterval = ni.classTwo.keySet.find(_.contains(dateP)).get
+        val lowerBound = selectedInterval.lowerValue.get
+        
+        val noOfWeeks = selectedInterval.numberOfWeeks.get
+        val response = List (
+          "Term Date" -> LocalDate.of(lowerBound.getYear, 4, 9), // unknown... but always the 9th of april
+          "Weekly Rate" -> data.weeklyRate.formatSterling,
+          "Rate Total" -> (data.weeklyRate * noOfWeeks).formatSterling
+        ) ++ (data.vdwRate match {
+          case None => Nil
+          case Some(vdw) => List (
+            "Voluntary Development Workers (VDW) Weekly Rate" -> vdw,
+            "Voluntary Development Workers (VDW) Total" -> vdw * noOfWeeks
+          )
+        }) ++ (data.shareFishingRate match {
+          case None => Nil
+          case Some(vdw) => List (
+            "Share Fishing Weekly Rate" -> vdw,
+            "Share Fishing Total" -> vdw * noOfWeeks
+          )
+        }) ++ List (
+          "Date Late For Short Term Benefits (STB)" -> "???",
+          "Final Date For Payment" -> lowerBound.plusYears(if (lowerBound.getYear < 1983) 3 else 7),
+          "Small Profits Threshold/Small Earnings Exemption (SPT/SEE)" ->
+            data.smallEarningsException.fold("???")(_.formatSterling), // below which paying is optional
+          "Date High Rate Provision (HRP) Applies" ->
+            (if (lowerBound.getYear < 1983) "" else lowerBound.plusYears(2).plusDays(1).toString),
+          "No of Wks" -> noOfWeeks.toString,
+          "Earnings Factor (includes enhance)" -> "???",
```

Comment:
I'm guessing the `???` are to fill in later right?

Reply:
yes. In the case of the earnings factor I'm not sure what 'enhance' means.